internal/telemetry/metrics/providers.go

```diff
@@ -73,32 +79,60 @@ func registerDefaultViews() error {
 
 // newProxyMetricsHandler creates a subrequest to the envoy control plane for metrics and
 // combines them with our own
-func newProxyMetricsHandler(promHandler http.Handler, envoyURL url.URL) http.HandlerFunc {
+func newProxyMetricsHandler(exporter *ocprom.Exporter, envoyURL url.URL, installationID string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		defer promHandler.ServeHTTP(w, r)
-
 		// Ensure we don't get entangled with compression from ocprom
 		r.Header.Del("Accept-Encoding")
 
-		r, err := http.NewRequestWithContext(r.Context(), "GET", envoyURL.String(), nil)
+		rec := httptest.NewRecorder()
+		exporter.ServeHTTP(rec, r)
+
+		err := writeMetricsWithInstallationID(w, rec.Body, installationID)
+		if err != nil {
+			log.Error().Err(err).Send()
+			return
+		}
+
+		req, err := http.NewRequestWithContext(r.Context(), "GET", envoyURL.String(), nil)
```

Comment:
I suppose it makes sense to add the request context (`r.Context()`) to any log entry in this function as well?

Reply:
I don't think I understand. What do we want out of the context in the log entry?